services/Donate.js

```diff
@@ -0,0 +1,112 @@
+const Donation = require("../models/Donation");
+
+class Donate {
+  /**
+   * @description create a new Flutterwave transaction
+   * @param {object} data
+   */
+  static async makeDonation(details) {
+    const url = "https://api.flutterwave.com/v3/payments";
+    const headers = {
+      Authorization: `Bearer ${process.env.FLW_SECRET_KEY}`,
+      "Content-Type": "application/json"
+    };
+    const fetchOptions = {
+      method: "POST",
+      headers,
+      body: JSON.stringify(details)
+    };
+
+    try {
+      const response = await fetch(url, fetchOptions);
+      const data = await response.json();
+
+      if (data.status === "error") {
+        return {
+          success: false,
+          error: data.errors
+        };
+      }
+
+      return {
+        success: true,
+        data
+      };
+    } catch (err) {
+      console.log(err.code);
+      console.log(err.response.body);
+    }
+  }
+
+  /**
+   * @description record a new transaction in db
+   * @param {object} data
+   */
+  static async recordTransaction(data) {
+    const newTransaction = new Donation({ ...data });
+    await newTransaction.save();
+
+    return newTransaction;
+  }
+
+  /**
+   * @description verify transaction status
+   * @param {string} tx_ref
+   */
+  static async verifyTransaction(tx_ref, expectedAmount, expectedCurrency) {
```

Comment:
Why do we need this extra verification step? I thought we only receive validated events from Flutterwave via the webhook?

Reply:
On the Flutterwave website, they said that it would be best to verify that the transaction hasn't been manipulated.
It's in their docs [there](https://developer.flutterwave.com/docs/integration-guides/webhooks/#verifying-webhook-signatures).